### DISCUSSION TITLE
SAME54 Xplained Pro - Using EthernetI have a problem with the Ethernet communication on the SAM E54 Xplained Pro.

### DISCUSSION THREAD

**Discussion by ErRiemenschneider** [zephyrproject-rtos/zephyr#43644] [Q&A]
I would like to download a binary file from the internet, but this does not work.

I use the following sample from the zephyr repo:
- http_get

Using the provided example URLs the response data is always empty (no header; no data). 
Intressting part, is, if I choose an URL which has the content type text (for example wikipedia.com/index.html), 
the download works as expected. Currently I'm not using SSL. 
Debugger shows that the recv() call returns length 0 which causes the while loop to exit.

Allready checkt:
- Diffrent URLs / Servers
- Diffrent Network
- Skipping DNS
- Skipping DHCP
- Build for QEMU (worked)

Thanks for your help

**Comment by jukkar** [comment]
The `samples/net/sockets/http_get` sample app probably should not to be used as is, first of all it just has 1kb buffer and not checking various corner cases etc. The `http_client` sample app is a bit more useful app to use as it can be modified more easily for real life scenarios. Both of these apps are more or less examples and normally used in network stack testing. You probably need to tweak them for your use cases.
